### ISSUE TITLE
Create rule S6853 (`jsx-a11y/label-has-associated-control`): label elements should have a text label and an associated control 

### ISSUE THREAD

**Issue by ilia-kebets-sonarsource** [SonarSource/SonarJS#4384]
[jsx-a11y/label-has-associated-control](https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/fffb05b38c8eee926ee758e9ceb9eae4e697fbdd/docs/rules/label-has-associated-control.md#jsx-a11ylabel-has-associated-control)

[RSPEC PR](https://github.com/SonarSource/rspec/pull/3445)

[implementation PR](https://github.com/SonarSource/SonarJS/pull/4396)
